File: apps/api/app/schemas/order_history.py
"""Contracts for explainable order-history intelligence."""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class BrandScore(BaseModel):
    brand: str
    score: float = Field(ge=0, le=1)


class CategoryScore(BaseModel):
    category: str
    score: float = Field(ge=0, le=1)


class PreferredBudget(BaseModel):
    min: int
    max: int


class OrderHistoryIntelligence(BaseModel):
    status: Literal["ok"]
    shoppingPersona: str
    favoriteBrands: list[BrandScore]
    favoriteCategories: list[CategoryScore]
    preferredBudget: PreferredBudget
    averageSpend: int
    shoppingFrequencyDays: int | None
    preferredSizes: dict[str, str]
    returnRate: float = Field(ge=0, le=1)
    recommendationReasons: list[str]


class InsufficientOrderHistory(BaseModel):
    status: Literal["insufficient_data"]
    ordersAnalyzed: int
    minimumRequired: int


OrderHistoryResponse = OrderHistoryIntelligence | InsufficientOrderHistory

File: apps/web/src/types/orderHistory.ts
export type RankedBrand = { brand: string; score: number };
export type RankedCategory = { category: string; score: number };

export type PreferredBudget = { min: number; max: number };

export type OrderHistoryIntelligence = {
  status: "ok";
  shoppingPersona: string;
  favoriteBrands: RankedBrand[];
  favoriteCategories: RankedCategory[];
  preferredBudget: PreferredBudget;
  averageSpend: number;
  shoppingFrequencyDays: number | null;
  preferredSizes: Record<string, string>;
  returnRate: number;
  recommendationReasons: string[];
};

export type InsufficientOrderHistory = { status: "insufficient_data"; ordersAnalyzed: number; minimumRequired: number };
export type OrderHistoryResponse = OrderHistoryIntelligence | InsufficientOrderHistory;
